src/net/http_serde.rs

```diff
@@ -541,49 +501,41 @@ pub mod query {
                         self.step.as_ref()
                     ))
                     .build()?;
-                let headers = Headers {
-                    offset: self.offset,
-                };
+                // TODO: this is a misuse of the origin header, and is insecure.
+                // need to authenticate clients with TLS.
```

Comment:
Do we have an issue to track this?

Reply:
Filed #597 (I was already planning to resolve this as part of the TLS work, which is partly done, but is currently on hold pending getting the server back to working condition).